### ISSUE TITLE
Cost Function과 Activation Function은 무엇인가요?

### ISSUE THREAD

**Issue by jarammm** [AInsteins/AI-Tech-Interview#12]
[3차시] 2022년 8월 30일 화요일

**Comment by kimyuuuuuu** [comment]
### Cost Function
- 비용함수 (=Loss Function : 손실 함수)

- 예측값과 데이터 값의 차이에 대한 함수.  원래의 값과 가장 오차가 작은 가설함수를 도출하기 위해 사용되는 함수.
- 이 가설함수의 형태를 결정짓는 것은 매개변수(parameter)라 부르는 **θ**이다.
    ⇒ 이 θ값을 적절하게 조정하여 **실제값 y에 가장 근접한 가설함수**를 Training set을 이용하여 도출해야 합니다.
    
- 신경망이 최적의 가중치 즉, 최적의 parameter(매개변수)를 찾게 만드는 지표이다. 이 loss function의 값이 클수록 신경망이 적절한 parameter를 찾지 못하는 것이다. 
⇒ 우리는 loss function의 값을 낮추기 위해 노력해야 한다.

- MSE, Cross Entropy 등

### Activation Function
- 활성화 함수

- **입력 신호의 총합을 출력 신호로 변환**하는 함수. **입력 신호의 총합이 다음 레이어의 활성화를 일으킬지 말지를 결정**한다.
- 비선형 함수인 활성화 함수가 선형 함수와 결합됨으로써 선형 모델은 비선형 모델이 된다.
- 선형 모델은 깊게 쌓을 수 없다. 깊게 쌓아도 하나의 층을 잘 튜닝한 것과 다르지 않기 때문이다.
- 비선형 모델은 깊게 쌓을 수 있다. 선형으로 만들었다가 비선형으로 만드는 작업을 계속 반복할 수 있기 때문이다. 이로 인해 모델은 복잡한 데이터에 대해 더 표현력이 좋아질 수 있다.
- 활성화 함수는 입력 값에 대해 더 높게 혹은 더 낮게 만들 수 있기 때문에 활성화 함수라고 불린다.

- Sigmoid, ReLU 등

**Comment by miinkang** [comment]
- 실제 값과 모델의 예측 값의 차이에 대한 함수를 cost function(비용 함수)이라고 합니다. 이 값을 낮추기 위해 파라미터를 조절할 수 있습니다. 대표적인 예로는 mse, mae 등이 있습니다. 

-   활성화 함수는 이름 처럼 이전 레이어를 통과한 값의 총 합을 출력 신호로 변환하는 함수입니다. 데이터를 잘 예측하기 위해는 복잡한 데이터를 처리할 수 있는 비선형 모델이 필요하고 선형 모델을 비선형으로 만드는 함수가 activation function입니다. 활성화 함수에는 sigmoid, relu 등이 있습니다.
